packages/assets-controllers/src/NftController.ts

```diff
@@ -1146,56 +1246,91 @@ export class NftController extends BaseController<NftConfig, NftState> {
    *
    * @param address - Hex address of the NFT contract.
    * @param tokenId - The NFT identifier.
+   * @param networkClientId - The networkClientId that can be used to identify the network client to use for this request.
    */
-  async addNftVerifyOwnership(address: string, tokenId: string) {
+  async addNftVerifyOwnership(
+    address: string,
+    tokenId: string,
+    networkClientId?: NetworkClientId,
+  ) {
     const { selectedAddress } = this.config;
-    if (!(await this.isNftOwner(selectedAddress, address, tokenId))) {
+    if (
+      !(await this.isNftOwner(
+        selectedAddress,
+        address,
+        tokenId,
+        networkClientId,
+      ))
+    ) {
       throw new Error('This NFT is not owned by the user');
     }
-    await this.addNft(address, tokenId);
+    await this.addNft(address, tokenId, { networkClientId });
   }
 
   /**
    * Adds an NFT and respective NFT contract to the stored NFT and NFT contracts lists.
    *
-   * @param address - Hex address of the NFT contract.
+   * @param tokenAddress - Hex address of the NFT contract.
    * @param tokenId - The NFT identifier.
-   * @param nftMetadata - NFT optional metadata.
-   * @param accountParams - The chain ID and address of network and account to which the nftContract should be added.
-   * @param source - Whether the NFT was detected, added manually or suggested by a dapp.
+   * @param options - an object of arguments
+   * @param options.nftMetadata - NFT optional metadata.
+   * @param options.chainId - The chain ID of the current network.
+   * @param options.userAddress - The address of the current user.
+   * @param options.source - Whether the NFT was detected, added manually or suggested by a dapp.
+   * @param options.networkClientId - The networkClientId that can be used to identify the network client to use for this request.
    * @returns Promise resolving to the current NFT list.
    */
   async addNft(
-    address: string,
+    tokenAddress: string,
     tokenId: string,
-    nftMetadata?: NftMetadata,
-    accountParams?: AccountParams,
-    source = Source.Custom,
+    {
+      nftMetadata,
+      chainId, // TODO remove and replace chainId arg with fetch chainId using getNetworkClientById(networkClientId).configuration.chainId once polling refactor is complete
+      userAddress,
+      source,
+      networkClientId,
+    }: {
+      nftMetadata?: NftMetadata;
+      chainId?: Hex; // TODO remove
+      userAddress?: string;
+      source?: Source;
+      networkClientId?: NetworkClientId;
+    } = { source: Source.Custom },
```

Comment:
a little bit awkward here. It's now possible to pass in an object without `source` set and source will not default to `Source.Custom` like the old behavior

Reply:
Agreed its a bit awkward 🤔. 

So I've added a tweak so that now when calling both of the methods to which `sources` is subsequently passed we fallback to the `Source.Custom` default if `source` is falsy: https://github.com/MetaMask/core/pull/1698/commits/613283d3a4c7c78a90440459bb589bf3adad3440

Reply:
actually nvm I've refactored a bit so now if the options object is present but `source` is left out it will default to `Source.Custom` and it will still default to `Source.Custom` if no options object is passed

Reply:
i believe the default value can just be empty object now? Good solution!

Reply:
done here: https://github.com/MetaMask/core/pull/1698/commits/83f5cb3a3c9e68e11ea96f4825c6545edc484312